pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -416,7 +416,7 @@ func garbageCollection(ctx context.Context, initShootClients func() (bool, error
 }
 
 func retrieveSeedConditions(ctx context.Context, botanist *botanistpkg.Botanist) ([]gardencorev1beta1.Condition, error) {
-	if botanist.ShootedSeed == nil {
+	if botanist.ManagedSeed == nil {
```

Comment:
I took a closer look (as I should have done it in the first place). Clearly this function and the logic that adds the seed conditions to the shoot conditions if the shoot is registered as a seed should be removed from the shoot care controller. However, I am not exactly sure what should replace it, perhaps because I don't fully understand why it exists in the first place.
* Should we add the seed conditions to the ManagedSeed conditions, and if yes, why? 
* If yes, can we avoid having a special "care" controller for that?

Reply:
@timebertt do you have suggestions how to "easily" achieve the removal of this code here? The `ManagedSeed` controller introduced by @stoyanr does not periodically reconcile, hence, it cannot manage this, or? Would we actually want this periodic reconciliation (does it have any side effects caused by the potentially periodic gardenlet deployment?)?

Anyways, what's the problem with this code here, i.e., why can't we simply keep it (at least for now unless there is an obvious way how to improve it)? It just fetches the `Seed`'s conditions and appends them to the `Shoot`'s conditions.

Reply:
Ok, I would envision the following:
The `ManagedSeed` controller watches `Seed` objects, enqueues them if it belongs to a `ManagedSeed` and the conditions have changed. Then it updates the conditions on the `Shoot` object. This would be the "idiomatic controller way", I guess, and there would be no need for a regular periodic sync.
Also, I think all controllers should be constructed in a way, that reconciliation operations are idempotent; in this case that would mean to not rollout the gardenlet if the spec hasn't changed (deployed `Deployment` spec won't change).

Though, I agree, that we can keep it for now as it is, to not delay the PR too much.

Reply:
OK, I will keep it then for now, perhaps with a TODO comment to improve it in a later PR.